Two Tutorials Added: Simulating Illumina Reads and Uploading Data to Cavatica

abhijna opened the pull request:
## PR Checklist

- [x] Release label added
- [x] PR category label added
- [x] PR mergeable
- [x] Clean build logs
- [x] Linked related issues
- [ ] Colorblindness test https://www.toptal.com/designers/colorfilter/

## PR Description
This is a new tutorial about using InSilicoSeq to simulating fastq data from a user-specified version of the human reference genome. It also includes some QC steps to see how the simulation looks. While this tutorial is good for simulating fastq reads from normal human samples, it cannot simulate mutations or indels.

Need someone to run through everything and make sure 1) that 16GB is enough for it all. 2) that it all works. I haven't done the formatting yet. If there are things you notice and can change easily, please do. If not, leave me a comment and I can fix it when I get to it.

Also added the command line uploader tutorial here. Was going to make a new PR but I pushed changes to this branch by mistake. Sorry :( 
Same deal for this tutorial as well. Check if everything run, look for typos, formatting problems, other ways to improve it. 

**Preview link**
https://cfde-training-and-engagement--263.com.readthedocs.build/en/263/Bioinformatics-Skills/Simulate_Illumina_Reads/

## Review format
Direct editing or comments are ok. 

## Timeline
This is due for December release. Don't care when you give me detailed comments as long as you give me enough time to incorporate your changes. If you can run through the code and let me know if it all installs on your computer in the next two weeks, that would be helpful.

s-canchi committed:
temporary folder name change for bioinfo

s-canchi committed:
Update Bioinfo-Tutorials to Bioinfo-Skills

s-canchi committed:
fixed .pages

s-canchi committed:
temporary folder name change for general

s-canchi committed:
Update general-temp to Web development

s-canchi committed:
update .pages

s-canchi committed:
update resources to temp folder

s-canchi committed:
update resources to Cheat sheets

s-canchi committed:
updating releases to temp folder

s-canchi committed:
Updating Releases to Release-Notes

s-canchi committed:
fixing .pages'

abhijna committed:
kids first moved

s-canchi committed:
Move CFDE portal tutorial to Bioinfo'

s-canchi committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

s-canchi committed:
fixed .pages

s-canchi committed:
Move conda to bioinfo

s-canchi committed:
move AWS to Bioinfo skills

s-canchi committed:
fix index and .pages for web develp

s-canchi committed:
fixes to bioinfo index and .pgaes

marisalim committed:
update title to cheat sheets

s-canchi committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

s-canchi committed:
temp disable of htmlproofer

s-canchi committed:
fixing .pages in Bioinfo skills

abhijna committed:
kf links fixed

abhijna committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

marisalim committed:
fixed cheat sheet links in snakemake, conda tutorials

marisalim committed:
fix snakemake, blast broken links

abhijna committed:
mime type and web dev links fixed

s-canchi committed:
broken links and title updates

s-canchi committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

s-canchi committed:
fixed broken links in cfde portal

s-canchi committed:
fixed broken links in protected branch tut

abhijna committed:
index pages

abhijna committed:
index fixes

abhijna committed:
release notes index fixed

abhijna committed:
index fix

marisalim committed:
cfde internal index pg title edit

s-canchi committed:
fixed titles in web devlp and added cat in index

s-canchi committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

marisalim committed:
cheat sheet title edits

abhijna committed:
index

marisalim committed:
fix spacing web devo index

abhijna committed:
fixes

abhijna committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

marisalim committed:
fix web devo pg

abhijna committed:
toc titles

abhijna committed:
Merge branch 'sam-nav-reorg' of https://github.com/nih-cfde/training-and-engagement into sam-nav-reorg

abhijna committed:
fixes

abhijna committed:
sim tutorial added

abhijna committed:
Merge branch 'dev' into abhijna

abhijna committed:
fixes

abhijna committed:
some fixes

abhijna committed:
new tutorial: cavatica uploader

abhijna commented:
Maybe Saranya can review the uploader tutorial and Marisa can review the InSilicoSeq? Sorry people. I really should have made two PRs but my brain did a stupid.

marisalim commented:
@abhijna here you go! got everything to work. I tried with 8Gb of storage, but it wasn't enough (the temp files take up a lot of space i guess). everything worked with the `t2.xlarge` and 16Gb of storage. time=22 minutes for `iss` command, total time to run tutorial is ~1hr, cost=...? (it doesn't show up on my billing dashboard yet) --> $0.18 :)
I tried the whole tutorial on the in-browser terminal. worked fine :)

here are things to edit/add:
### formatting things
- [x]  needs the tutorial landing page info - estimated time, prerequisites, learning objectives. you could put the "Additional Resources" into the "Tutorial Resources" section
- [x] use backticks for AWS instance names and command flags (e.g., `--model`, `--mode basic`)
- [x] update step number format ("Step 1" instead of 1)
- [x] images need captions in the link syntax

### InSilicoSeq tutorial
#### Intro
- [x] maybe give examples of what the answer is (e.g., variants?)/ how 'knowing the answer' helps: "With simulated data, the user knows the final answer"
#### setting up AWS instance
- [x] need info/warning that the t2.xlarge machine is not part of the free tier and will cost some money (add estimated cost and time to run. maybe add to estimated time section of landing page info. speaking of which, how much does the GWAS tutorial cost if you don't have free tier access anymore? we should add a cost estimation if it's not there already)
- [x] AWS tutorial link to the overview page instead of vidlet?
- [x] after choosing t2.xlarge, click on the "4. Add Storage" (e.g., can skip step 3, don't have to click on the 'next:' button)
- [x] say what the 16Gb is for - e.g., the output simulated data files we're generating are x Gb (actually, the outputs are small, it looks like the temp files take up a lot of space)
#### InSilicoSeq
- [x] add link to software page for more info and other options: https://github.com/HadrienG/InSilicoSeq, https://insilicoseq.readthedocs.io/en/latest/. maybe put the additional resources earlier OR link down to them
- [x] when you do the commands below
```
sudo apt update
sudo apt install python3-pip
```
it asks if you want to:
```
1 upgraded, 50 newly installed, 0 to remove and 32 not upgraded.
Need to get 46.8 MB of archives.
After this operation, 200 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```
tell users if they should enter "Y" (same thing for the fastqc installation command)
- [x] there are a bunch of warnings from the insilicoseq installation - maybe add a note that it's ok to ignore these
```
Installing collected packages: pysam, numpy, biopython, scipy, future, joblib, InSilicoSeq
  WARNING: The scripts f2py, f2py3 and f2py3.8 are installed in '/home/ubuntu/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts futurize and pasteurize are installed in '/home/ubuntu/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script iss is installed in '/home/ubuntu/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed InSilicoSeq-1.5.1 biopython-1.78 future-0.18.2 joblib-0.17.0 numpy-1.19.4 pysam-0.16.0.1 scipy-1.5.4
```
- [x] add an `iss --version` check to make sure it's installed properly (also helps if they find they're using a different version from the one in the tutorial)
- [x] say the downloaded human genome file should be 3.1 Gb when uncompressed
- [x] move the "it takes approximately 40 minutes to run" up to the top of the tutorial along with cost estimate
- [x] `--compress true` error (resolved by deleting: this flag works sometimes and sometimes not -_-)
```
iss generate --genomes ../reference_genome/GRCh38_latest_genomic.fna --model hiseq --n_reads 5M --cpus 4 --output my_sim --compress true
usage: iss [subcommand] [options]
InSilicoSeq: error: unrecognized arguments: true
```
- [ ] can you run stuff on AWS and then walk away? (e.g., use screen or tmux) or do you need to keep it open for 40 mins
- outputs:
    - [ ] what do the abundance numbers mean from "my_sim_abundance.txt"?
    - [x] my output files are called "my_sim_R1.fastq" and "my_sim_R2.fastq" (not "my_sim_1.fastq")
    - [ ] could show how to check out the fastq files (`less -S my_sim_R1.fastq"). why are the sequences a mix of lower and upper case letters? (does it matter?)
    - [x] the 5M reads are split between the R1 and R2 (instead of creating 5M each). also I don't have exactly 2.5M per read direction, why? as a simulator, shouldn't it be exact? total, I have 4,999,694 reads total between the R1 and R2 files
- [x] for the for loop command, add that you could do this to simulate data for a bunch of samples otherwise users might not know what to do with that code
- [x] for the warning, can recommend starting out with a very small set of reads to test the code before scaling up
#### Quality control
- [x] edit this, the outputs will not be compressed unless you add a `gzip` step (assuming the `--compress` flag is deleted from `iss` command)
```
cd fastqc <- i think this is supposed to be the fastq directory. there's no fastqc directory
fastqc *.gz <- should we gzip the files? otherwise it will be `fastqc *.fastq`
```
- [x] local transfer - reminder to do this in local terminal and not AWS terminal
- [x] the backticks got included in this code block:
```
scp -i ~/Desktop/amazon.pem ubuntu@ec2-??-???-???-??.us-east-2.compute.amazonaws.com:/home/ubuntu/fastq/\*.html ~/Desktop/fastqc/.
```
- [x] need to type "yes" for local download "Are you sure you want to continue connecting (yes/no/[fingerprint])? yes"
- [x] my fastqc outputs say the read lengths are 126bp not 125bp

#### end of tutorial
- [x] remember to say how to close the instance! and local download the fastq files if they want to keep them.

abhijna commented:
Thanks, Marisa!! Working on it. A couple of questions (will add more if they come up):

> update step number format ("Step 1" instead of 1)

Do you mean for this section: "Selecting the right AWS instance to run InSilicoSeq"?

--> This is completely up to you how to organize sections into steps. :) I was thinking that the steps under the AWS section could be "### Step 1: Select instance", "### Step 2: Add additional storage", etc. so they show up in the right-side TOC.

> images need captions in the link syntax

Aren't my images captioned? e.g. ![t2 xlarge](../images/Simulated_Data_t2xlarge.png)

--> the website doesn't display the text in the brackets for some reason so you have to put them in quotes, e.g.,:
![t2 xlarge](../images/Simulated_Data_t2xlarge.png "t2 xlarge")

abhijna commented:
Documentation about abundance.txt doesn't make sense to me. posted an issue about it: https://github.com/HadrienG/InSilicoSeq/issues/196

abhijna commented:
There seems to be an open issue about why you get fewer reads than specified in the insilicoseq command

https://github.com/HadrienG/InSilicoSeq/issues/195

abhijna committed:
incorporated marisa's changes

abhijna committed:
Merge branch 'dev' of https://github.com/nih-cfde/training-and-engagement into abhijna

abhijna committed:
minor formatting

s-canchi commented:
On my radar and will work on it first thing on Monday!

abhijna committed:
added learning obs etc

abhijna committed:
formatting

s-canchi committed:
Added text and code for updating the instance

s-canchi committed:
Corrected curl download links 

The original curl commands did not work and gave a "Could not resolve host" error. I added the correct syntax that worked along with clarificatons to the commands explanation. I also added the link to the reference paper.

s-canchi committed:
Added -y flag to ensure no user input

s-canchi committed:
Added code to check java installation success

s-canchi committed:
Added $PATH steps and creating alias

s-canchi committed:
Minor details for auth token step

s-canchi committed:
Minor text corrections, steps annotation

s-canchi committed:
minor typo

s-canchi committed:
Added -y flag

s-canchi committed:
minor typo correction

s-canchi committed:
minor format corrections

s-canchi committed:
minor text addition

s-canchi reviewed (commented):
@abhijna - I made some corrections and additions to the data uploader tutorial directly (since I couldn't continue with the tutorial otherwise) but I also have additional comments:

- [x]  In the introduction I am a bit confused by this line: “Cavatica provides….”. Is this tutorial an extension of the listed methods or a better version of the available methods ? Maybe rephrase the sentence to clarify.

- [x] The learning objectives list "Learn to edit metadata of files on Cavatica". But those steps are not present in the current tutorial. Will you be adding more to it ?

- [ ] In Step 2,  for the dowloaded files, would it make sense to add the md5sum check steps ? [Original steps in ANGUS](https://angus.readthedocs.io/en/2019/quality-and-trimming.html)

- [x] In Step 3, could you add some explanation for the different flags used in the tar command.

- [ ] Consider adding some screenshots for the position of username, auth token and the files tab for Cavatica. Alternatively you can also consider adding a screen recording.

abhijna commented:
Thanks Saranya! I'll work on these in the next couple of days. I thought I had already added the edit metadata sets. I guess not :'(

abhijna commented:
So thoughts:

1) on md5sum check: for the KF tutorial, the nature of the file we are using in the transfer doesn't really matter. So we don't need to do an md5sum after downloding into AWS. At the Cavatica end, I'm not sure how to do it on their interface :(. Any ideas?
2) >Consider adding some screenshots for the position of username, auth token and the files tab for Cavatica. Alternatively you can also consider adding a screen recording.

I didn't want to add anything here because the" portal set up and permissions tutorial" walks people through finding the authentication token with screenshots. I don't want to have duplicates -- it will be a pain to edit the same thing in two places when the screenshots need to get updated. I've linked that page, though.  Thoughts?

abhijna committed:
minor edits

abhijna committed:
metadata editing section added

abhijna committed:
typos fixed

abhijna commented:
@s-canchi I added the metadata editing section. Could you take a look to make sure it's ok? Should be pretty straightforward I think.

s-canchi commented:
> on md5sum check: for the KF tutorial, the nature of the file we are using in the transfer doesn't really matter. So we don't need to do an md5sum after downloding into AWS. At the Cavatica end, I'm not sure how to do it on their interface :(. Any ideas?

This is tricky indeed not just for this tutorial but for data in general. So I think of two ways to get this information in Cavatica. 
- Approach one would be verify the md5sum of the files on AWS after download from osf and add the md5sum as a metadata point for the downloaded file to show in the Cavatica interface. This however assumes that the data file integrity is sustained during the upload from AWS to Cavatica. 
- Approach two is to create a CLI in Cavatica using interactive analysis option and to run the md5sum on the downloaded files. I tested this approach and it works. The negative with this approach is that it involves further explanation of steps to set up the interactive analysis. 

I leave the final option to you of whether you want to include these steps or not. But I assume for future tutorials we might want to have some sort of sanity check for data files. 

> I didn't want to add anything here because the" portal set up and permissions tutorial" walks people through finding the authentication token with screenshots. I don't want to have duplicates -- it will be a pain to edit the same thing in two places when the screenshots need to get updated. I've linked that page, though. Thoughts?

That's a valid point. Thinking as a user without much familiarity of the Cavatica interface, I like the visual impact of the screenshots with the circled sections, such that when I go back to the Cavatica interface I can go by visual memory instead of text. Clicking back and forth between two separate tutorials may not be ideal as we want to keep them modular, and hence my suggestion a screen-capture video which can be relatively quick to accomplish.  
I will leave the choice to you about how you would like to incorporate any visual aides. This can also be an enhancement feature that can be tackled next time :)

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+
+## Selecting the right AWS instance to run InSilicoSeq
+
+To run [InSilicoSeq](https://github.com/HadrienG/InSilicoSeq), you will need to launch a 64 bit Ubuntu Server 20.04 LTS (HVM), SSD Volume Type instance as described in [the AWS tutorial](Introduction_to_Amazon_Web_Services/introtoaws3.md) with a **few modifications**:
```
let's rephrase this - it's possible to run InSilicoSeq locally on Mac or an HPC. it does not need to be AWS. 

"In this tutorial, we will run InSilicoSeq on a 64 bit Ubuntu Server 20.04 LTS (HVM), SSD Volume Type instance, as described in..."

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+To run [InSilicoSeq](https://github.com/HadrienG/InSilicoSeq), you will need to launch a 64 bit Ubuntu Server 20.04 LTS (HVM), SSD Volume Type instance as described in [the AWS tutorial](Introduction_to_Amazon_Web_Services/introtoaws3.md) with a **few modifications**:
+
+ ### Step 1)
+ You must select the the `t2.xlarge` instance (instead of the default t2.micro instance selection described in the tutorial):
```
format: there are few more places where `t2.micro` and `t2.xlarge` need the backtick formatting

the `###` steps are not showing up in the right-side TOC for some reason. also, nitpicky thing: `### Step 1:` colon instead of `)` please :)

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+
+## Why use simulated data?
+
+There are a myriad of bioinformatics tools currently available to users. Before use, every tool must be tested for general usability and suitability for a given task. Simulated data can be ideal for the purposes of testing bioinformatics tools because the user can model the data set based on their specified parameters. With simulated data, the user knows the final answer, and so they know when th ...
```
this got a little complicated - how about something like this?

> With so many bioinformatics tools currently available to users, simulated datasets are ideal for quickly generating toy datasets based on real data to test new tools for general usability and user task suitability. For example, a researcher could test a variant calling pipeline on data simulated from the human genome. If the data simulation did not introduce new mutations relative to the original human genome used for simulation, variants should not be detected. Simulated datasets are also useful for teaching, since they can be readily shared unlike protected (human) data."

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
````diff
@@ -0,0 +1,337 @@
+    ```
+    sudo apt update
+    sudo apt install python3-pip
+    ```
````
format: add an empty line between the tabbed box sections, the `sudo` code isn't showing up on the website. same thing for AWS Instance Code boxes below

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
````diff
@@ -0,0 +1,337 @@
+    cd fastq
+    ```
+
+Finally, run the code to make your simulated fastq file.
````
oops, let's put back the time estimate for this particular step - "The simulation step takes ~30 minutes to complete. (something about leave the terminal open until it's completed...otherwise it may interrupt the run?)"

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+    `--output` specifies the name of the output files. You're calling it my_sim, but you can change it to whatever suits your needs.
+
+
+If your run is successful, you will see two ".fastq" files, "my_sim_R1.fastq" and "my_sim_R2.fastq", and an abundance file called "my_sim_abundance.txt". The abundance file is more meaningful in the context of simulating metagenomic data. It is a tab-delimited file containing the abundance of each genome supplied to the command. The 5 M reads generated per simulation are split between the R1 and  ...
```
oh, so if you simulated data from multiple genomes this would tell you the proportion from each genome in the simulated set? confusing that with 1 input genome, there are so many lines in the output abundance txt file..

i don't know what to do about the imprecise number of simulated reads (since the bug issue on the insilicoseq repo hasn't been answered. i suppose you could add to the thread to say it's a problem for us too). you could add a note that it generates approximately the requested number, but not exactly...?

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+
+## Looping InSilicoSeq
+
+If you want to repeat the simulation multiple times, modify the following code block by changing the `n` to an integer corresponding to the number of time you wish to run the simulation. Your output files will be named 1_my_sim_R1 and 1_my_sim_R2, 2_my_sim_R1 and 2_my_sim_R2, ..., n_my_sim_R1 and n_my_sim_R2.
```
minor edits: 

If you want to simulate data for multiple samples, modify the following code block by changing the `n` to the number of samples you want to simulate. Your output files will be named:

-  "1_my_sim_R1.fastq" and "1_my_sim_R2.fastq"
-  "2_my_sim_R1.fastq" and "2_my_sim_R2.fastq"
- ...
-  "n_my_sim_R1.fastq" and "n_my_sim_R2.fastq"

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
````diff
@@ -0,0 +1,337 @@
+    ```
+
+!!! Warning
+    Depending on how many times you run the simulation and/or how many reads you simulate, the code may take a really long time to complete! You can start out by running a small set of reads (2-3) to test the code before scaling up.
````
"small set of reads (`--n_reads 1M`)" something like this to show which flag to edit

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+
+### Interpretation
+
+**Watch this [YouTube video](https://www.youtube.com/watch?v=bz93ReOv87Y) on how to interpret FastQC results**:
```
Add in a reference that this video was made by the Babraham Bioinformatics Institute (the developers of FastQC).

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+
+[![FastQC video tutorial](http://img.youtube.com/vi/bz93ReOv87Y/0.jpg)](https://www.youtube.com/watch?v=bz93ReOv87Y "FastQC Interpretation")
+
+All simulated reads are 126 bp long :)
```
edit suggestion: "From the FastQC report, we see that the overall read quality scores are good and read length is 126bp. These simulated reads are now ready to be used for analysis!"

marisalim commented on docs/Bioinformatics-Skills/Simulate_Illumina_Reads.md:
```diff
@@ -0,0 +1,337 @@
+
+    - To avoid unnecessary charges, remember to [terminate your AWS instance](Introduction_to_Amazon_Web_Services/introtoaws4.md) once you are done using it.
+
+    - It is important to remember that terminating the instance causes all the data in that instance to be lost forever. If you wish to keep the fastq files, please remember to transfer them to your local machine or analysis platform. Check out our [tutorial on transferring files from AWS to Cavatica](Kids-First/Upload_Data.md).
```
hm, is there a reason to move these fastq files to Cavatica? vs. the more general instructions for downloading to local computer? (also because not everyone doing this tutorial will have the login set up to Cavatica)

marisalim reviewed (commented):
@abhijna looking good! here are a few more edits. it would be good to get 1 more person to run through the tutorial!

abhijna commented:
Ok, I made all the changes. Thank you :)

 You're right, I think we need one more person to run through the tutorial to see if it all makes sense and runs correctly. Maybe @josesanchez1815 or @jeremywalter

abhijna committed:
marisa's changes incorporated

abhijna committed:
screencast edits

josesanchez1815 commented:
I'll take a look

s-canchi committed:
Added edit metadata to learning objectives

jeremywalter commented:
TOC in Illumina looks wonky because it just says
Step 1: 
Step 2: 
Step 3:
Step 4: 

How about -->
Step 1: Select Instance
Step 2: Add Storage
Step 3: Review and Launch
Step 4: Access the Instance

(also... I'm doing this ^^^)

s-canchi commented on docs/Bioinformatics-Skills/Kids-First/Upload_Data.md:
```diff
@@ -0,0 +1,382 @@
+
+![](images/Edit_Metadata_Button.png "Edit Metadata Button")
+
+- You should see a pop-up window on the right side of the screen
```
It seems that Cavatica has set metadata fields that you can edit. Is there a way to add new metadata terms if not listed in the options ?

abhijna replied:
I'm 98% sure that the answer is no, but I will double check.

abhijna replied:
nevermind there is a roundabout way of doing this. significant addition coming up.

marisalim replied:
@abhijna i can test this tutorial when you're done adding this metadata part

abhijna replied:
postponed to the next release. it's not working right now. will fuss with it some at a later time.

jeremywalter committed:
grammar edits and punctuation

jeremywalter commented:
Thoughts and observations (Illumina Reads): 

1) I did not run through the tutorial but the bottom steps seems like they might trip me up  -- I'm betting the transfer to local computer part might be confusing for others (as it would be potentially for me)... paths and different terminals are confusing! I'm not sure how to fix that without a screencast. When I clicked on the link (path to amazon.pem) I felt more confused.

2) At the end it warns users to turn off their instance. I suggest putting part or all of this in the intro. Why? Because when I ran the AWS long ago I got e-mails about my usage for weeks and weeks and finally turned it off after the lesson we just had. Why? Maybe I didn't finish the AWS tutorial... sometimes people don't work their way to the end. Since I was using free-tier nothing happened to me, but it's a good warning to preface any cloud lesson with.. 'don't forget to turn it off when you're done or $$$' 

Note: I'm just here looking for what's wrong... but I love all the 'enter this code' and 'here's the output' steps... :)

abhijna commented:
@jeremywalter thank you! this is helpful. I will rework it.

josesanchez1815 commented:
This is form the Simulating Illumina reads

HI! I think the only comments I have for this tutorial is I got confused about when to use the fastqc command? 
It was an unsuccessful run, because I wasn't sure which folder to be in.

So, when after the command 
`sudo apt install fastqc` 
I enter `cd fastq`

and I created a new empty folder with no fastqc files and tried to run fastqc there. Obviously, it didn't do anything because I did not have any files.

So I think maybe omit `cd fast q` because we are already inside the fastq folder at this point.

jeremywalter commented:
![image](https://user-images.githubusercontent.com/40613731/101423196-62062680-38ad-11eb-9256-b60f0c87f740.png)
This section from uploading to Cavatica is curiously absent of explanation (compared to the rest of the lesson). What's tar? what's the -C flag and the mysterious ~?

jeremywalter committed:
grammar edits and punctuation for upload_data lesson

abhijna committed:
minor edits

abhijna committed:
fixes

abhijna committed:
jose and jeremy's comments incorporated

abhijna committed:
link to alternative method of metadata edit

abhijna committed:
Merge branch 'dev' into abhijna

abhijna commented:
I made all the changes. Put two of the comments (about md5sum and screencasts) into issues for the next release.

abhijna committed:
local hosting of vidlets

jeremywalter committed:
change header formatting (caps)

jeremywalter committed:
change header formatting

s-canchi committed:
updating relative link for cast files

jeremywalter committed:
change header formatting

s-canchi committed:
correction to relative link to cast files

jeremywalter committed:
header format for blast tutorial (caps)

jeremywalter committed:
header format for BLAST (caps)

jeremywalter commented:
RE: minor edits commit -- I already did that? could create a conflict? leaving this here for reference in case it does (how did that happen?)

marisalim committed:
small text edits - specify t2.micro, formatting

marisalim commented:
@abhijna for the Cavatica uploader tutorial:
- all the steps worked!
- minor: the cavatica uploader is a ".tgz" file, but the explanation of the `tar` command talks about "tar.gz" extension. are they the same? ---> yes it is the same. I changed it to say .tgz :) 
- this can be an edit for next iteration, but i think it would be best to stick to 1 example dataset - show the yeast example files in the Cavatica screenshots instead of the sim data ones. ---> yeah, i'd have to redo all the screenshots. so this can be a thing for next time. 
- add a note/link to close down the AWS instance at the end of the tutorial. ---> done

abhijna committed:
termination warning added

abhijna commented:
Done :)

abhijna committed:
Merge branch 'abhijna' into abhijna_vidlet_edits

abhijna committed:
Merge pull request #285 from nih-cfde/abhijna_vidlet_edits

Edits to KF Data Download Screencasts

abhijna committed:
Merge branch 'dev' into abhijna

marisalim reviewed (approved):
looks like all the changes i requested got added! good to go
